### ISSUE TITLE
about val_acc with scikit_learn.train_test_split

### ISSUE THREAD

**Issue by guofuzheng** [keras-team/keras#11869]
Hello,  i met a problem. I test my code with the train and test samples specified in  dataset and i get accuracy_1 (for example). I test my code with train and test samples splitted by train_test_split and i get accuracy_2 (for example), and the number of test samples is same as the number of test samples specified in dataset.  But the gap between accuracy_1 and accuracy_2 is very large. accuracy_2 is higher than accuracy_1. I am troubled by this. Thank you for your help.

**Comment by liebscher** [comment]
It's difficult to say without seeing the data, but it's obvious that your first test set is from a different distribution than the training set. Is your second example a split of the training and the testing sets combined from the first example? If so, by shuffling and then splitting the data you'd be getting a more even exposure to the distributions of the data and thus higher testing accuracy.

If I were you, I would explore and visualize the training and testing sets in the first example to see how they differ. Then, if your problem constrains you to using a specific test set, you could adjust your training model accordingly.
